src/EnergyPlus/CondenserLoopTowers.cc

```diff
@@ -5826,44 +5840,52 @@ namespace CondenserLoopTowers {
             ShowContinueError(
                 state,
                 "Design inlet air wet-bulb or approach temperature must be modified to achieve an acceptable range at the minimum air flow rate");
             ShowContinueError(state, format("Cooling tower simulation failed to converge for tower {}", this->Name));
-            //    if SolFla = -2, Tr is returned as minimum value (0.001) and outlet temp = inlet temp - 0.001
-        } else if (SolFla == -2) {    // decide if should run at max flow
-            Real64 TempSetPoint(0.0); // local temporary for loop setpoint
-            switch (state.dataPlnt->PlantLoop(this->plantLoc.loopNum).LoopDemandCalcScheme) {
-            case DataPlant::LoopDemandCalcScheme::SingleSetPoint: {
-                TempSetPoint = state.dataPlnt->PlantLoop(this->plantLoc.loopNum).LoopSide(this->plantLoc.loopSideNum).TempSetPoint;
-            } break;
-            case DataPlant::LoopDemandCalcScheme::DualSetPointDeadBand: {
-                TempSetPoint = state.dataPlnt->PlantLoop(this->plantLoc.loopNum).LoopSide(this->plantLoc.loopSideNum).TempSetPointHi;
-            } break;
-            default: {
-                assert(false);
-            } break;
-            }
-            if (this->WaterTemp > (TempSetPoint + state.dataCondenserLoopTowers->towers(this->VSTower).MaxRangeTemp)) { // run flat out
-                OutletWaterTempLocal = this->WaterTemp - state.dataCondenserLoopTowers->towers(this->VSTower).MaxRangeTemp;
+        } else if (SolFla == -2) {
+            // bad starting value means that solution corresponds to a range that is beyond
+            // the bounds of the model; The maximum range is used here
+            outletWaterTempLocal = this->WaterTemp - this->MaxRangeTemp;
+            ++this->VSErrorCountTRCalc;
+            if (this->VSErrorCountTRCalc < 2) {
+                ShowWarningError(
+                    state,
+                    format("The range for the cooling tower {} likely exceeds the bounds of the model. The maximum range of the model is used {}.",
+                           this->Name,
+                           this->MaxRangeTemp));
+                ShowContinueError(state,
+                                  format(" ... Occurrence info = {}, {} {}",
+                                         state.dataEnvrn->EnvironmentName,
+                                         state.dataEnvrn->CurMnDy,
+                                         General::CreateSysTimeIntervalString(state)));
+            } else {
+                ShowRecurringWarningErrorAtEnd(
+                    state,
+                    format("The range for the cooling tower {} likely exceeds the bounds of the model. The maximum range of the model is used {}",
+                           this->Name,
+                           this->MaxRangeTemp),
+                    this->ErrIndexTRCalc);
```

Comment:
Previously, if a "bad starting value" error was returned by Regula Falsi (`SolFla == -2`, i.e., both the output for the function for the lower and upper bounds are of the same sign), the outlet temperature was only recalculated if the inlet temperature minus the range were greater than the setpoint. As described in the issue, this occurs for the `YorkCalc` model. Looking at that model closer shows that this typically occurs when the range returned by the model is significantly greater than the maximum range. In the first figure below, the solution is a range of about 63 deg. C which is way beyond the limit of the model (22.2222).

Inlet water temperature: 38.75 deg. C; Wet-bulb temperature: 15.4 deg. C; Flow ratio: 2
![image](https://github.com/NREL/EnergyPlus/assets/5149279/cecb47ab-629c-42b9-87f6-90b0948e80cd)

Varying wet-bulb temperature
![image](https://github.com/NREL/EnergyPlus/assets/5149279/737484c0-4f17-4f37-844d-354a97692a55)

Varying air flow ratio
![image](https://github.com/NREL/EnergyPlus/assets/5149279/86e35707-764c-471f-9c38-411d893e5d17)

Varying water flow ratio
![image](https://github.com/NREL/EnergyPlus/assets/5149279/e42f38cf-8d55-4879-85fe-a46fb4ae7bc4)

Increasing further the upper bound used with Regula Falsi works but doesn't make sense since the solutions are outside the limits of the model. The proposed changes apply the maximum range every time `SolFla == 2`. As shown in the simulation results above, this approach leads to the correct control of the outlet of the tower and reasonable fan power unlike the status quo.

Currently, when the outlet is not recalculated, no warning is issued, The proposed changes include a (recurring) warning every time this happens.